.grit/patterns/math_prod.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Use `math.prod` instead of multiplying in a loop
+---
+
+This pattern transforms a loop that computes the product of a list of numbers into a call to `math.prod` (introduced in Python 3.8).
+
+
+```grit
+engine marzano(0.1)
+language python
+
+pattern prod_init($accum) {
+    or {
+        `$accum = 1`,
+        `$accum = 1.0`,
+    }
+}
+
+pattern prod_accum($accum, $factor) {
+    or {
+        `$accum *= $factor`,
+        `$accum = $accum * $factor`,
+        `$accum = $factor * $accum`,
+    }
+}
+
+`$for` where {
```

Comment:
This will match *anything* and then restrict it, which ends up being much worse for performance. It's better to hoist the `for_statement` up as the main thing we're matching against.

Reply:
Done.